src/Symfony/Component/Mailer/Envelope.php

```diff
@@ -44,6 +44,10 @@ public static function create(RawMessage $message): self
 
     public function setSender(Address $sender): void
     {
+        // to ensure deliverability of bounce emails independent of UTF-8 capabilities of SMTP servers
+        if (!preg_match('/^[^@\x80-\xFF]++@/', $sender->getAddress())) {
+            throw new InvalidArgumentException(sprintf('Invalid sender "%s": non-ASCII characters not supported in local-part of email.', $sender->getAddress()));
+        }
```

Comment:
I think we should also change `DelayedEnvelope::getSender()` like this:

```
    public function getSender(): Address
    {
        if (!$this->senderSet) {
            parent::setSender(self::getSenderFromHeaders($this->message->getHeaders()));
        }

        return parent::getSender();
    }
```

So that we also get the exception when the sender in the envelope is not set explicitly but comes from the email headers.

Reply:
@fabpot done but 2 test fails seem a bit odd:

```
There were 2 failures:

1) Symfony\Component\Mailer\Tests\EnvelopeTest::testSenderFromHeaders
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
 Symfony\Component\Mime\Address Object (
     'address' => 'return@symfony.com'
-    'name' => 'return'
+    'name' => ''
 )

/var/www/symfony/src/Symfony/Component/Mailer/Tests/EnvelopeTest.php:71

2) Symfony\Component\Mailer\Tests\EnvelopeTest::testSenderFromHeadersWithoutFrom
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
 Symfony\Component\Mime\Address Object (
     'address' => 'from@symfony.com'
-    'name' => 'from'
+    'name' => ''
 )
``` 

So the name of the address is lost now. This seems due to 

https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Mailer/Envelope.php#L47

where the name is ignored. Is that correct?

Reply:
Yes, it is correct. The sender can only be an email address.